.github/workflows/ci_action.yml

```diff
@@ -90,6 +90,9 @@ jobs:
           pip install -e .[DEV,ML]
           pip install gdal==$(gdal-config --version)
 
+      - name: Set up local cluster
+        run: ray start --head
```

Comment:
this is now also necessary to do before running tests and from a researcher/user point-of-view I would expect that the testing ray cluster would be set up in the conftest or something.

I'm now starting to get worried that I got prematurely excited and helped push us in this direction 😬 let's hope these are just temporary obstacles which need a small paradigm shift

Reply:
it's simple, you expect too much :P

the debugging part is IMO very crucial, and i'll try to resolve it. Without it this change should not yet happen (or we should at least include a 'debug' mode which runs it with an EOExecutor in  a single thread)

Reply:
As discussed, there are still things to be discussed. I'm fine with this MR being merged and we can make updates on top of it, or we wait with the merging until we figure out the logging and storage manager stuff, in case that affects the pipelines